src/plugin.json

```diff
@@ -3,6 +3,7 @@
   "type": "app",
   "name": "Logs",
   "id": "grafana-lokiexplore-app",
+  "preload": true,
```

Comment:
Do we need `preload:true` when exposing components? (I used to think this was for non-hook component extensions only)

Reply:
yeah, I think so. It doesn't work for me without `preload: true`.

Reply:
Interesting, might need some insights from @leventebalogh, also whether we need to [specify the extension in the plugin.json](https://github.com/grafana/grafana-pdc-app/blob/4c81a7ea864b72f4e7ce7580c2d957795a2215b1/src/plugin.json#L116)

Reply:
please take a look at this thread before proceeding this preloading:

https://raintank-corp.slack.com/archives/C077S1MP40M/p1719937543634869

Reply:
Thanks @leeoniya, looks like no need for `preload: true` is supported in 11.1.0, and it requires the plugin.json metadata as mentioned by @sympatheticmoose in the thread: 
> we'd tried a metadata extractor approach which proved unviable, but theres alternatives by **just having the information manually added to plugin.json** etc

Reply:
@jarben @leeoniya 

**Unfortunately we currently need the `preload: true` flag.** It's true, that [we are working on only loading plugins when they are used](https://github.com/grafana/grafana/pull/86624) (extensions or normal plugin usage), but the [PR](https://github.com/grafana/grafana/pull/86624) is not yet merged and deployed.

> ideally we can avoid introducing this rather than trying to undo it later.

Yeah, I agree that this would be _ideal_, however unfortunately we cannot avoid it for now. Regarding undoing it later: once the [PR mentioned above](https://github.com/grafana/grafana/pull/86624) is merged `preload: true` won't have an effect, which means that even though we still need to do clean ups in the `plugin.json`, it won't affect performance.

> we need to [specify the extension in the plugin.json](https://github.com/grafana/grafana-pdc-app/blob/4c81a7ea864b72f4e7ce7580c2d957795a2215b1/src/plugin.json#L116)

Yes, that would be great, thanks. (We can only preload plugins properly if we know what they are exposing.)

Reply:
Thanks @leventebalogh ! So is this PR just [a prerequisite](https://github.com/grafana/grafana/pull/83085)? Or is it for Grafana extensions only, not plugin extensions? 

> once the https://github.com/grafana/grafana/pull/86624 is merged preload: true won't have an effect, which means that even though we still need to do clean ups in the plugin.json, it won't affect performance.

Makes me think that the `cloud-home-app` also uses [preload:true](https://github.com/grafana/hg-cloud-home-admin/blob/c0af6423d49de53d73b97fca2abc488709ee4aca/src/plugin.json#L61) to be able to [show notification banners](https://github.com/grafana/hg-cloud-home-admin/blob/c0af6423d49de53d73b97fca2abc488709ee4aca/src/preload.ts#L14) anywhere in Grafana, not just after the plugin is loaded. How would we handle this use case if `preload:true` doesn't work anymore?

Reply:
Thanks @leventebalogh, great to see the cloud-home-app exception was considered in the PR. 
Do you remember which library caused the issue? @grafana/cloud-ab-testing?

Reply:
Exactly, it's the third-party library that it is using for A/B testing. (The issue is that it's somehow causing a re-rendering of the whole `AppChrome` in case it's not loaded before Grafana boots up. I was trying to debug it, but couldn't find the root cause during a time-box.)